### PR TITLE
arch: Support more platforms

### PR DESCRIPTION
**Pull request by mloubout**
Generalizes gcc
suport graviton 2-3-4 and add cflags to use gcc with it as well

### CODE REVIEW

File: devito/arch/compiler.py [hunk 1]
```diff
@@ -434,6 +434,10 @@ def __init_finalize__(self, **kwargs):
         if platform in [POWER8, POWER9]:
             # -march isn't supported on power architectures, is -mtune needed?
             self.cflags = ['-mcpu=native'] + self.cflags
+        elif platform is Graviton:
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:437]: it shuold be `isinstance(platform, Graviton)` no?

File: devito/arch/compiler.py [hunk 2]
```diff
@@ -434,6 +434,10 @@ def __init_finalize__(self, **kwargs):
...
+            # Graviton flag
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:438]: Drop useless comment

File: devito/arch/compiler.py [hunk 3]
```diff
@@ -434,6 +434,10 @@ def __init_finalize__(self, **kwargs):
...
+            mx = platform.march
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:439]: probably also inlinable

File: devito/arch/compiler.py [hunk 4]
```diff
@@ -462,8 +466,9 @@ def __init_finalize__(self, **kwargs):
         platform = kwargs.pop('platform', configuration['platform'])
 
         # Graviton flag
-        if platform is GRAVITON:
-            self.cflags += ['-mcpu=neoverse-n1']
+        if platform is Graviton:
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:469]: `isinstance(platform, Graviton)`

File: devito/arch/compiler.py [hunk 5]
```diff
@@ -962,7 +967,26 @@ def __new_with__(self, **kwargs):
...
+    """
+
+    def __getitem__(self, key):
+        if key.startswith('gcc-'):
+            i = key.split('-')[1]
+            return partial(GNUCompiler, suffix=i)
+        return super().__getitem__(key)
+
+    def has_key(self, k):
+        return k in self.keys() or k.startswith('gcc-')
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:983]: no need for "()" after `keys`

**Review by FabioLuporini — CHANGES_REQUESTED** [review]

**Comment by FabioLuporini** [comment]
I've also changed the PR title to start with "arch: ..." rather than "compiler: ..."

**Review by georgebisbas — APPROVED** [review]

**Comment by ggorman** [comment]
Reinstating Graviton runner so this can actually be tested.

**Review by ggorman — APPROVED** [review]

File: devito/arch/compiler.py (hunk 1 above)
Comment (mloubout) [reply devito/arch/compiler.py]: Fixe

File: devito/arch/compiler.py (hunk 5 above)
Comment (mloubout) [reply devito/arch/compiler.py]: Yes you do

**Review by FabioLuporini — APPROVED** [review]
GTG now

**Pull request merged**